docs/developer-env.md

```diff
@@ -0,0 +1,473 @@
+(developer-env)=
+
+# Environment setup
```

Comment:
I think it'd be good to have an admonition / note thing saying "this is once you've installed Nextflow, click here for instructions on that". So that if anyone drops directly onto this page they're not confused.

Reply:
Added extra line.